templates/zerver/app/home.html

```diff
@@ -8,6 +8,14 @@
             </g>
         </svg>
     </div>
+    <div class="history-limited-box">
+        <p>
+            {% trans %}
+            More messages are older than your organization's 10,000 message history limit.
+            <a href="/plans/">See plans and pricing</a>
+            {% endtrans %}
+        </p>
```

Comment:
I tweaked this to make clear that we definitely know there are more messages.  And also to links to /plans/, since that has information.

Reply:
(Oh, and to tag for translation!)

Reply:
I find the wording a bit awkward here for some reason ("More messages are older...").  I would say something like "There are more messages, but they are too old for your org's, limit" or something like that.